spec/array_spec.rb

```diff
@@ -0,0 +1,48 @@
+require 'array'
+
+describe Array do
+  subject {[1,2,3,4]}
+  describe 'faisal_inject' do
+
+    it 'adds' do
+      expect(subject.faisal_inject{|k,v| k + v}).to eq 10
+    end
+
+    it 'adds with an argument' do
+      expect(subject.faisal_inject(10){|k,v| k + v}).to eq 20
+    end
+
+    it 'adds with an argument' do
+      expect(subject.faisal_inject(10){|k,v| k * v}).to eq 240
```

Comment:
Space missing after comma.<br>Space missing to the left of {.<br>Space between { and | missing.<br>Space missing inside }.